mesonbuild/backend/backends.py

```diff
@@ -832,7 +832,9 @@ def determine_rpath_dirs(self, target: T.Union[build.BuildTarget, build.CustomTa
     @staticmethod
     @lru_cache(maxsize=None)
     def canonicalize_filename(fname: str) -> str:
-        parts = Path(fname).parts
+        if mesonlib.is_windows():
+            fname = fname.replace('\\', '/')
+        parts = fname.split('/')
```

Comment:
This function is already cached by fname, so this change will only ever affect one call per filename that gets canonicalized. Is Path() actually showing up in profiling here?

Reply:
In its performance improvement tracking issue, @bonzini was mentioning speed up object_filename_from_source . The `canonicalize_filename` call was using a significant time in that function, and pathlib stuff was the culprit. 

That being said, `object_filename_from_source` is not really a hotpath when generating my project. I could drop that commit if you think it doesn't worth it.

Reply:
When profiling my project, about half of the time in `object_filename_from_source` is spent in `relpath`, and maybe 20% in `canonicalize_filename`. In `canonicalize_filename`, half of the time is spent in `Path.parts`, and 20% in `Path.__init__`. I'm not sure I ever hit the `sha1` call here. 

Globally, I got 670 call to `canonicalize_filename`, and 27 calls to `sha1`, and time spent in `sha1` is almost nothing.